### ISSUE TITLE
"Untethered" logging for Android

### ISSUE THREAD

**Issue by siebert** [httptoolkit/httptoolkit#224]
Hi,

I would really like the option to log the HTTPS requests on my rooted Android device while it's not connected to my PC running HTTP Toolkit to be able to do traces in specific locations with the Android device alone without having to carry a notebook.

The workflow should be something like this:
- I connect my rooted Android device via USB to my PC and select "Android device via ADB" as usual
- I remove the USB cable from my Android device, breaking the adb connection.
- The HTTP Toolkit App on the Android device notices the adb disconnect and automatically starts the "record" mode (or I do select it manually in the App if the automatic detection isn't possible)
- The HTTPS traffic is recorded and stored on the Android phone by the App
- At a later time, I reconnect the Android device via USB to my PC
- HTTP Toolkit on the PC detects the reconnected Android device, downloads the recorded traffic from the App and shows it a usual.
---

_Does this affect you too? Click the +:smiley: icon in the top right and add a :+1: on issues to help decide where HTTP Toolkit goes next, or [go take a look & vote on the most popular ideas so far](https://github.com/httptoolkit/feedback/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc)._

**Comment by pimterry** [comment]
I agree this would be useful, but for now it's probably out of scope.

Today, the Android code doesn't actually know anything about HTTP or HTTPS at all. All it does is connect to the computer and configure the device certificates, and then it redirects traffic at the raw TCP packet level. That means it would be a huge amount of work to intercept HTTP on the device without the computer attached - it effectively requires rebuilding the whole app from scratch for Android!

That said, there might be a route here to avoid that in theory... The core app can be run as a headless node.js application with the UI connecting remotely from elsewhere, and it's [apparently](https://code.janeasystems.com/nodejs-mobile) possible to run node.js on Android. That means if you could run [httptoolkit-server](https://github.com/httptoolkit/httptoolkit-server/) on your Android device, then that would be capable of handling all the HTTP proxying & interception logic to do on-device capture and mocking using the code we already have, and then you could connect your computer to that as a UI, to view the traffic and manage the rules remotely.

This is basically the same thing proposed for in-container Docker interception in #122, and it has other interesting use cases too, like running HTTP Toolkit instances in the cloud that you can attach to with a local UI, to create a headless publicly accessible debugging proxy.

_But_ anyway, that's still quite a bit of work, and I'm not 100% sure it's possible regardless, especially since I've never actually tested Node on Android.

In the meantime there are probably workarounds though. Have you considered exposing the proxy port (i.e. 8000, by default) publicly to the internet, and proxying your traffic that way? That would require manual proxy configuration I think, but given that your mobile device could send traffic via HTTP Toolkit on your computer as a proxy even when it's running elsewhere in the world, as long as it has internet connectivity. You'd have to leave HTTP Toolkit running somewhere to do this, but that might be workable for you?
